crates/cargo-platform/src/cfg.rs

```diff
@@ -89,6 +89,19 @@ impl CfgExpr {
             CfgExpr::Value(ref e) => cfg.contains(e),
         }
     }
+
+    pub fn fold<E>(&self, f: &impl Fn(&Cfg) -> Result<(), E>) -> Result<(), E> {
```

Comment:
This doesn't align with how `fold` is used in [`Iterator::fold`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold) and found that confusing when trying to understand this.

It might also be good to have a comment explaining the intended semantics.